web/src/beta/lib/reearth-ui/components/TextArea/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { FC, useCallback, useEffect, useState, ChangeEvent, useRef } from "react";
+
+import { fonts, styled } from "@reearth/services/theme";
+
+export type TextAreaProps = {
+  value?: string;
+  placeholder?: string;
+  resizable?: "none" | "height";
+  disabled?: boolean;
+  rows?: number;
+  counter?: boolean;
+  maxLength?: number;
+  onChange?: (text: string) => void;
+  onBlur?: (text: string) => void;
+};
+
+export const TextArea: FC<TextAreaProps> = ({
+  value,
+  placeholder,
+  resizable,
+  rows,
+  disabled,
+  counter,
+  maxLength,
+  onChange,
+  onBlur,
+}) => {
+  const [currentValue, setCurrentValue] = useState(value ?? "");
+  const [isFocused, setIsFocused] = useState(false);
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    setCurrentValue(value ?? "");
+  }, [value, resizable]);
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLTextAreaElement>) => {
+      const newValue = e.currentTarget.value;
+      setCurrentValue(newValue);
+      onChange?.(newValue);
+    },
+    [onChange],
+  );
+
+  const handleBlur = useCallback(() => {
+    setIsFocused(false);
+    onBlur?.(currentValue);
+  }, [currentValue, onBlur]);
+
+  const handleFocus = useCallback(() => {
+    setIsFocused(true);
+  }, []);
+
+  return (
+    <Wrapper>
+      <TextAreaWrapper status={isFocused ? "active" : "default"}>
+        <StyledTextArea
+          resizable={resizable}
+          ref={textareaRef}
+          rows={rows ? rows : 3}
+          value={currentValue}
+          disabled={disabled}
+          placeholder={placeholder}
+          onChange={handleChange}
+          onBlur={handleBlur}
+          onFocus={handleFocus}
+          maxLength={maxLength}
+        />
+      </TextAreaWrapper>
+      {counter && maxLength ? (
+        <CharacterCount>
+          {currentValue.length} / {maxLength}
+        </CharacterCount>
+      ) : (
+        counter && <CharacterCount>{currentValue.length}</CharacterCount>
+      )}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled("div")(() => ({
+  display: "flex",
+  flexDirection: "column",
+}));
+
+const TextAreaWrapper = styled("div")<{
+  status: "default" | "active";
+}>(({ theme, status }) => ({
+  border:
+    status === "default" ? `1px solid ${theme.outline.weak}` : `1px solid ${theme.select.main}`,
+  borderRadius: theme.radius.small,
+  background: theme.bg[1],
+  display: "flex",
+  boxShadow: theme.shadow.input,
+}));
+
+const StyledTextArea = styled.textarea<{ resizable?: "none" | "height"; disabled?: boolean }>(
+  ({ theme, resizable, disabled }) => ({
+    outline: "none",
+    border: "none",
+    background: "none",
+    resize: resizable === "height" ? "vertical" : "none",
+    overflow: resizable === "height" ? "hidden" : "auto",
```

Comment:
Do we need this hidden? I think it should have a scroll bar when resizable is height and the content is a long text.